app/controllers/authentication_controller.rb

```diff
@@ -0,0 +1,12 @@
+class AuthenticationController < ApplicationController
+  def token_auth
+    token = params[:auth_token]
+    payload = decode(token)
+    if User.where('email = ?', payload[:email])
+      render json: {in_new: false, token: token}
+    else
+      User.create!(name: payload[:name], email: payload[:email], google_token: token)
+      render json: {is_new: true, token: token}
+    end
```

Comment:
All nice! Need to learn about this OAuth stuff...

Reply:
So, to try and summarize here: if the user matches the email given and is not `in_new` we just send a response with the token back. If alternatively, they are a match and `in_new` we create the new user and also send the response with the token. Does that check out? Just trying to make sure I'm following here.

Reply:
in_new was a typo, is fixed now. 
if we find the email in our user database, then they are not a new user. we send the token and is_new: false
if we don't find the email in our db, then they are a new user. we send the token and is_new: true